pkg/monitors/mongodb/atlas/monitor.go

```diff
@@ -0,0 +1,149 @@
+package atlas
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/signalfx/signalfx-agent/pkg/monitors/mongodb/atlas/measurements"
+
+	"github.com/signalfx/signalfx-agent/pkg/utils"
+
+	"github.com/Sectorbob/mlab-ns2/gae/ns/digest"
+	"github.com/mongodb/go-client-mongodb-atlas/mongodbatlas"
+	"github.com/signalfx/golib/v3/datapoint"
+	"github.com/signalfx/signalfx-agent/pkg/core/config"
+	"github.com/signalfx/signalfx-agent/pkg/monitors"
+	"github.com/signalfx/signalfx-agent/pkg/monitors/types"
+
+	"github.com/signalfx/signalfx-agent/pkg/utils/timeutil"
+)
+
+func init() {
+	monitors.Register(&monitorMetadata, func() interface{} { return &Monitor{} }, &Config{})
+}
+
+// Config for this monitor
+type Config struct {
+	config.MonitorConfig `yaml:",inline"`
+	// ProjectID is the Atlas project ID.
+	ProjectID string `yaml:"projectID" validate:"required" `
+	// PublicKey is the MongoDB Atlas public API key
+	PublicKey string `yaml:"publicKey" validate:"required" `
+	// PrivateKey is the MongoDB Atlas private API key
+	PrivateKey string `yaml:"privateKey" validate:"required" neverLog:"true"`
+	// Timeout for HTTP requests to get MongoDB Atlas process measurements. This should be a duration string that is accepted by https://golang.org/pkg/time/#ParseDuration
+	Timeout timeutil.Duration `yaml:"timeout" default:"5s"`
+	// EnableCache is a flag to enable caching
+	EnableCache bool `yaml:"enableCache" default:"true"`
```

Comment:
Expand this description.  I have no idea what the cache does or the implications of using it.